salt/fileclient.py

```diff
@@ -142,12 +142,18 @@ def _cache_loc(self, path, saltenv='base', cachedir=None):
                                     path)
         destdir = os.path.dirname(dest)
         cumask = os.umask(63)
-        if not os.path.isdir(destdir):
-            # remove destdir if it is a regular file to avoid an OSError when
-            # running os.makedirs below
-            if os.path.isfile(destdir):
-                os.remove(destdir)
+
+        # remove destdir if it is a regular file to avoid an OSError when
+        # running os.makedirs below
+        if os.path.isfile(destdir):
+            os.remove(destdir)
+
+        # ensure destdir exists. Handle OSError in case it is already there
+        try:
             os.makedirs(destdir)
+        except OSError as e:
+            pass
```

Comment:
I would still quack about that into logs why this had happened. I still believe that muting all that completely isn't best, because you might have another reason why that had happened and _maybe_ you want to see it. So I would:
```python
try:
    os.makedirs(destdir)
except OSError as err:
    e_code, e_msg = err.args
    if e_code != 17:  # Other than already exists
        log.error("Unable to create a directory: '{message}' (exit code: {code})".format(
            message=e_msg, code=e_code)
```